src/ray/core_worker/transport/direct_task_transport.cc

```diff
@@ -53,6 +53,17 @@ Status CoreWorkerDirectTaskSubmitter::SubmitTask(TaskSpecification task_spec) {
                                                   push_task_reply,
                                                   reply.actor_address(),
                                                   /*is_application_error=*/false);
+            } else if (status.IsCreationTaskError()) {
+              RAY_LOG(DEBUG) << "Actor creation failed and we will not be retrying the "
+                                "creation task, actor id = "
+                             << actor_id << ", task id = " << task_id;
+              // Copy the actor's reply to the GCS for ref counting purposes.
```

Comment:
```suggestion
              // Copy the actor's reply to the GCS for ref counting purposes.
              // The actor failure is handled from GCS, so we treat it as a task being completed.
              // We don't use  `FailOrRetryPendingTask` here because actor creation retry is handled by GCS.  
```